### ISSUE TITLE
Incompatible types with types/react v18

### ISSUE THREAD

**Issue by Josh-Cena** [rpearce/image-zoom#302]
This issue is a bug report.

## Steps to reproduce behavior

1. Install `@types/react@18.0.0`
2. Write some code importing this library:

```tsx
import React, { type ReactNode } from "react";
import BasicZoom from "react-medium-image-zoom";
import "react-medium-image-zoom/dist/styles.css";

export default function Zoom({ children }: { children: ReactNode }): JSX.Element {
  return <BasicZoom>{children}</BasicZoom>;
}
```

3. Run `yarn tsc`

## Expected behavior

This should compile

## Actual behavior

```
../node_modules/react-medium-image-zoom/dist/cjs/Controlled.d.ts(2,43): error TS2305: Module '"react"' has no exported member 'ReactType'.
```

The problem is that `ReactType` is removed in v18 in favor of `ElementType`. See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210

We can easily work around this by adding `"skipLibCheck": true`, but just would like you to know that such an issue exists. Moreover I noticed this lib is still using `@types/react@16` and `react@16`—any plans to upgrade? I can help with the process.

## Link to repository/example of behavior

https://github.com/facebook/docusaurus/runs/5952102932?check_suite_focus=true

**Comment by rpearce** [comment]
I have had plans wax and wane for this project for quite some time, and there are numerous branches that have come and gone with all sorts of upgrades and improvements that never saw the light of day.

I'll take a look at this tonight / tomorrow night if I can!

PRs are also welcome ❤️

**Comment by rpearce** [comment]
This is released under `v4.3.7`, and you've been added to the `AUTHORS` file and `All Contributors` section in the `README`.

I'll work on getting the type dev dependencies updated separately.

**Comment by Josh-Cena** [comment]
Great, thanks!
